sugar.go

```diff
@@ -0,0 +1,169 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package zap
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+// Sugar is a wrapper around core logger whith less verbose API
+type Sugar interface {
+	// Check the minimum enabled log level.
+	Level() Level
+	// Change the level of this logger, as well as all its ancestors and
+	// descendants. This makes it easy to change the log level at runtime
+	// without restarting your application.
+	SetLevel(Level)
+
+	// Create a child logger, and optionally add some context to that logger.
+	With(...interface{}) (Sugar, error)
+
+	// Log a message at the given level. Messages include any context that's
+	// accumulated on the logger, as well as any fields added at the log site.
+	Log(Level, string, ...interface{}) error
+	Debug(string, ...interface{}) error
+	Info(string, ...interface{}) error
+	Warn(string, ...interface{}) error
+	Error(string, ...interface{}) error
+	Panic(string, ...interface{}) error
+	Fatal(string, ...interface{}) error
+	// If the logger is in development mode (via the Development option), DFatal
+	// logs at the Fatal level. Otherwise, it logs at the Error level.
+	DFatal(string, ...interface{}) error
+}
+
+type sugar struct {
+	core Logger
+}
+
+// NewSugar is a constructor for Sugar
+func NewSugar(core Logger) Sugar {
+	return &sugar{core}
+}
+
+func (s *sugar) Level() Level {
+	return s.core.Level()
+}
+
+func (s *sugar) SetLevel(lvl Level) {
+	s.core.SetLevel(lvl)
+}
+
+func (s *sugar) With(args ...interface{}) (Sugar, error) {
+	fields, err := getSugarFields(args...)
+	if err != nil {
+		return nil, err
+	}
+	return NewSugar(s.core.With(fields...)), nil
+}
+
+func getSugarFields(args ...interface{}) ([]Field, error) {
+	if (len(args) % 2) != 0 {
+		return nil, errors.New("invalid number of arguments")
+	}
+	var (
+		fields []Field
+		ii     int
+		key    string
+		value  interface{}
+	)
+	for ii, value = range args {
+		if (ii % 2) == 0 {
+			switch value.(type) {
+			case string:
+				key = value.(string)
+			default:
+				return nil, errors.New("field name must be string")
+			}
+		} else {
+			switch value.(type) {
+			case bool:
+				fields = append(fields, Bool(key, value.(bool)))
+			case float64:
+				fields = append(fields, Float64(key, value.(float64)))
+			case int:
+				fields = append(fields, Int(key, value.(int)))
+			case int64:
+				fields = append(fields, Int64(key, value.(int64)))
+			case uint:
+				fields = append(fields, Uint(key, value.(uint)))
+			case uint64:
+				fields = append(fields, Uint64(key, value.(uint64)))
+			case string:
+				fields = append(fields, String(key, value.(string)))
+			case time.Time:
+				fields = append(fields, Time(key, value.(time.Time)))
+			case time.Duration:
+				fields = append(fields, Duration(key, value.(time.Duration)))
+			case fmt.Stringer:
+				fields = append(fields, Stringer(key, value.(fmt.Stringer)))
+			default:
```

Comment:
I'd prefer not to do this, since the common (non-error) case will then require passing a `nil`.

With this API, callers will need to pass both the key and the value for errors: `sugar.With("error", err)`. We'll need to add a case like this:

```
case error:
    fields = append(fields, String(key, value.(error).Error()))
```